packages/syft/src/syft/core/tensor/tensor.py

```diff
@@ -108,18 +109,43 @@ def simple_add(self, other: Any) -> TensorPointer:
 
         result_public_shape = None
 
-        if self.public_shape is not None and other.public_shape is not None:
+        op = getattr(operator, op_str)
+
+        if isinstance(other, TensorPointer):
+            other_shape = other.public_shape
+        elif isinstance(other, (int, float)):
+            other_shape = (1,)
+        elif isinstance(other, np.ndarray):
+            other_shape = other.shape
+        else:
+            raise ValueError(f"Invalid Type for TensorPointer:{type(other)}")
+
+        if self.public_shape is not None and other_shape is not None:
             result_public_shape = (
-                np.empty(self.public_shape) + np.empty(other.public_shape)
+                op(np.empty(self.public_shape), np.empty(other_shape))
             ).shape
 
         result.public_shape = result_public_shape
 
         return result
 
-    def __add__(self, other: Any) -> Union[TensorPointer, MPCTensor]:
+    @staticmethod
+    def _apply_op(
+        self: TensorPointer,
+        other: Union[TensorPointer, MPCTensor, int, float, np.ndarray],
+        op_str: str,
+    ) -> Union[MPCTensor, TensorPointer]:
+        """Performs the operation based on op_str
+
+        Args:
+            other (Union[TensorPointer,MPCTensor,int,float,np.ndarray]): second operand.
 
-        if self.client != other.client:
+        Returns:
+            Tuple[MPCTensor,Union[MPCTensor,int,float,np.ndarray]] : Result of the operation
+        """
+        op = getattr(operator, op_str)
+
+        if isinstance(other, TensorPointer) and self.client != other.client:
```

Comment:
Q: I know that this piece was already here, but do you know if this works in case we have:
`mpc_tensor1` between `client1` and `client2` and
`mpc_tensor2` between `client2` and `client3`?

Reply:
I created a notebook and tested the functionality george in the same PR,
Since we have the reconstruct problem, I thought to add hagrid test after a fix for that.
https://github.com/OpenMined/PySyft/blob/ch2858/notebooks/smpc/SMPC%20Tensor%20pointer%20Abstraction.ipynb

Reply:
I tried covering all the edge cases in the notebook, If there are any other edge cases comment , george, I will include it.

Reply:
The previous case did not work for MPCTensors george, only between tensor pointer and MPCTensor, with different subsets of parties,I think now it work for all scenario.